migrations/multiple_contract_migration.go

```diff
@@ -0,0 +1,230 @@
+package migrations
+
+import (
+	"fmt"
+	"github.com/onflow/cadence/runtime/ast"
+	"github.com/onflow/cadence/runtime/parser2"
+	"github.com/onflow/flow-go/engine/execution/state"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/rs/zerolog"
+	"strings"
+)
+
+type MultipleContractMigrationError struct {
+	Errors []error
+}
+
+func (e *MultipleContractMigrationError) Error() string {
+	var b strings.Builder
+	b.WriteString("multiple contract migration errors:")
+	for i, err := range e.Errors {
+		_, err := fmt.Fprintf(&b, "\nerr %d: %s", i, err)
+		if err != nil {
+			panic(err)
+		}
+	}
+	return b.String()
+}
+
+type MultipleContractMigration struct {
```

Comment:
does it have to be a struct? I didn't get the reason to use a cache, anyway the cache won't be persisted to file and each key value pair is returned only once.

Reply:
the idea was to return all the payloads to the same method at the same time so it can be stateless and build any data structure needed inside that migration method.

Reply:
I can change the migration instead of methods to 
> 
type Migration interface {
	Name() string
	Migrate(payload []Payload) ([]Payload, error)
}

if you think we need it

Reply:
The cache is intended as a small performance mechanism. 

There are multiple register with the same owner and once we have migrated the code for one owner we don't need to keep checking the other registers for the same owner. Thats the only thing the cache does. If the cache is lost it would only mean we would unnecessarily be checking more registers than needed.

Reply:
I was thinking about the upcoming migration for storage capacity and storage used. Two fields(registers) will be added to each account. To do that storage used needs to be calculated for each account by adding up all the register sizes per address. In the current format of migrations I can't know when I have encountered all registers from a single address. @ramtinms any idea on how to approach this?

Reply:
Can't we iterate twice and make a dictionary of owner to payloads ?

Reply:
We got this resolved. The storage fees contract will be able to iterate through the registers only once, then dump all the extra registers it created once it reached the end of `payload`